dotnet-desktop-guide/net/wpf/events/routed-events-overview.md

```diff
@@ -0,0 +1,260 @@
+
+The following example registers the `Tap` routed event, stores the returned `RoutedEvent` instance, and implements a CLR event wrapper.
+
+:::code language="csharp" source="./snippets/routed-events-overview/WpfControlLibraryCsharp/CustomButton.cs" id="RegisterRouteEventAndClrAccessors":::
+:::code language="vb" source="./snippets/routed-events-overview/WpfControlLibraryVb/CustomButton.vb" id="RegisterRouteEventAndClrAccessors":::
+
+### Routed event handlers and XAML
+
+To add an event handler in XAML, you declare the event name as an attribute on the event listener element. The attribute value is your handler method name. The handler must be implemented in the code-behind partial class for the XAML page. The event listener is the element where the event handler is attached and invoked. For an event that's a member (inherited or otherwise) of the listener class, you can attach a handler as follows:
+
+:::code language="xaml" source="./snippets/routed-events-overview/csharp/MainWindow.xaml" id="AddHandler_UnqualifiedEventName":::
+
+If the listener is a different element than the element that raised the event, it's possible that the event isn't a member of the listener's class. In that case, you must use the qualified event name, in the form of `<owner type>.<event name>`. For example, because the <xref:System.Windows.Controls.StackPanel> class doesn't implement the <xref:System.Windows.Controls.Primitives.ButtonBase.Click> event, you'll need to use the qualified event name:
+
+:::code language="xaml" source="./snippets/routed-events-overview/csharp/MainWindow.xaml" id="AddHandler_QualifiedEventName":::
+
+The XAML syntax for adding routed event handlers is the same as for standard CLR event handlers. For more information about adding event handlers in XAML, see [XAML in WPF](../xaml/index.md).
+
+> [!NOTE]
+> The element that originally raised a routed event is identified as the <xref:System.Windows.RoutedEventArgs.Source?displayProperty=nameWithType> in the event handler parameters. The event listener is the element where the event handler is attached and invoked, and is identified as the [sender](xref:System.Windows.RoutedEventHandler) in the event handler parameters.
```

Comment:
Do you think we should delete this and add any missing info the `## Attach and implement a routed event handler` section?

Reply:
Good idea! Done.